services/concept_insights/ci.js

```diff
@@ -0,0 +1,422 @@
+            var matches = res.matches;
+            for (var i=0; i<matches.length; i++) {
+              concepts.push(matches[i].id);
+            }
+            msg.concepts = concepts;
+            msg.payload = res;
+            node.send(msg);
+          }
+        });
+      });
+    });
+  }
+
+  RED.nodes.registerType('watson-concept-insights-search-concepts', searchConceptsNode, {
+    credentials: {
+      username: {type:"text"},
+      password: {type:"password"}
+    }
+  });
+
+  function relatedConceptsNode(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    this.on('input', function(msg) {
+      setupConceptInsightsNode(msg,config,this,function(concept_insights,account_id) {
+       
+        var params = {
+          graph: '/graphs/wikipedia/en-latest',
+          concepts: msg.concepts
+        }
+        node.status({fill:"blue",shape:"ring",text:"requesting..."});
+        concept_insights.graphs.getRelatedConcepts(params, function(err, res) {
+          node.status({});
+          if (err) {
+            var message = err.error;
+            return node.error(message, msg);
+          } else {
+            msg.payload = res;
+            node.send(msg);
+          }
+        }); 
+      });
+    }); 
+  }
+
+  RED.nodes.registerType('watson-concept-insights-related-concepts', relatedConceptsNode, {
+    credentials: {
+      username: {type:"text"},
+      password: {type:"password"}
+    }
+  });
+
+
+  function uploadDocumentNode (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    this.on('input', function (msg) {
+
+      setupConceptInsightsNode(msg,config,this,function(concept_insights,account_id) {
+        this.corpus = RED.nodes.getNode(config.corpus);
+        if (this.corpus) {
+          var corpus_name = this.corpus.cname;
+          var access = this.corpus.access;
+
+          if (this.corpus.publiccorpus) {
+            corpus_name = this.corpus.publiccorpus;
+            account_id = 'public';
+          }
+          var params = {
+            corpus: '/corpora/'+account_id+'/'+corpus_name,              
+          }
+
+          //If corpus doesn't exist, create it
+          concept_insights.corpora.getCorpus(params, function(err,res) {
+            if (err && err.error === 'not found') {
+              //Create corpus
+              params.access = access;
+              node.status({fill:"blue",shape:"ring",text:"Creating corpus "+corpus_name});
+              concept_insights.corpora.createCorpus(params, function(err,resp) {
+                if (err) {
+                  var message = "Creating Corpus error: "+err.error;
+                  return node.error(message, msg);
+                } else {
+                  addDocument(checkStatus(msg.payload.length));
+                }
+              });
+            } else {
+              node.status({fill:"blue",shape:"ring",text:"Using existing corpus "+corpus_name});
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2392368955/issues/source?bid=3225115&fileBranchId=3228814#l287)